### ISSUE TITLE
Ionide does not start

### ISSUE THREAD

**Issue by l1x** [ionide/ionide-vscode-fsharp#1663]
### Describe the bug

Invalid attempt to spread non-iterable instance. In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
Cannot read property 'subscriptions' of undefined

### Steps to reproduce

Open a Fsharp project.

### Link to sample reproduction

```fsproj
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>

    <AssemblyName>Depoxy</AssemblyName>
    <RootNamespace>DataDeft.Lambda</RootNamespace>

    <OutPutType>Exe</OutPutType>
    <TargetFramework>net6.0</TargetFramework>
    <LangVersion>latest</LangVersion>
	  <ImplicitUsings>enable</ImplicitUsings>
    <AWSProjectType>Lambda</AWSProjectType>
    <PublishTrimmed>true</PublishTrimmed>
	  <PublishReadyToRun>true</PublishReadyToRun>
    <AssemblyName>bootstrap</AssemblyName>
    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>

  </PropertyGroup>

  <ItemGroup>
    <Compile Include="List.fs" />
    <Compile Include="Seq.fs" />
    <Compile Include="Option.fs" />
    <Compile Include="Result.fs" />
    <!-- config -->
    <Compile Include="Config.fs" />
    <!-- lib -->
    <Compile Include="Hashing.fs" />
    <Compile Include="Utils.fs" />
    <Compile Include="Glue.fs" />
    <Compile Include="Ses.fs" />
    <Compile Include="S3.fs" />
    <Compile Include="ExternalId.fs" />
    <!-- init -->
    <Compile Include="Init.fs" />
    <!-- api:common -->
    <Compile Include="ApiCommon.fs" />
    <!-- api:specific -->
    <Compile Include="UserApi.fs" />
    <Compile Include="JobApi.fs" />
    <Compile Include="TableApi.fs" />
    <Compile Include="AggregateApi.fs" />
    <!-- handler -->
    <Compile Include="Handler.fs" />
    <!-- main -->
    <Compile Include="Main.fs" />

    <None Include="config\**\*.yaml">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    </None>

  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Amazon.Lambda.AspNetCoreServer" Version="6.1.0" />
    <PackageReference Include="Amazon.Lambda.Core" Version="2.1.0" />
    <PackageReference Include="Amazon.Lambda.Serialization.Json" Version="2.0.0" />
    <PackageReference Include="AWSSDK.Glue" Version="3.7.9.21" />
    <PackageReference Include="AWSSDK.SimpleEmailV2" Version="3.7.1.45" />
    <PackageReference Include="AWSSDK.S3" Version="3.7.5" />
    <PackageReference Include="FSharp.Data" Version="4.2.6" />
    <PackageReference Include="Falco" Version="3.1.7" />
    <PackageReference Include="FSharp.SystemTextJson" Version="0.17.4" />
    <PackageReference Include="LazyCache" Version="2.4.0" />
    <PackageReference Include="SimpleBase" Version="3.1.0" />
    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.14.1" />
    <PackageReference Include="System.Data.HashFunction.SpookyHash" Version="2.0.0" />
    <PackageReference Include="Microsoft.Extensions.Caching.Memory" Version="6.0.0" />
  </ItemGroup>

<ItemGroup>
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
</ItemGroup>

</Project>

```

### Expected behaviour

A working environment.

### Screenshots

![Screenshot 2022-02-02 at 14 59 30](https://user-images.githubusercontent.com/20079/152168010-8527cb06-c095-4ba0-ab6f-391dece04be5.png)

## Machine info
 - OS:Mac
 - .NET SDK version: [e.g. 6.0.100]
 - Ionide version: v5.10.2

**Comment by cartermp** [comment]
@l1x this:

```xml

<!-- <ItemGroup>
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
</ItemGroup>

</Project>
```

would mean that the project file is considered invalid because the closing `</Project>` tag is commented out. Can you uncomment that part?

**Comment by l1x** [comment]
Why did you close this? The file paste was the accident not this bug.

**Comment by cartermp** [comment]
Since I don't have any source files that match your project, I had to remove them - but I can load this project without issue otherwise, and Ionide recognizes all the package references and whatnot. Do you have a minimal reproduction that can be shared that doesn't involve removing all the file items?

I'm using the same .NET SDK and Ionide version on macOS.

**Comment by l1x** [comment]
> Since I don't have any source files that match your project, I had to remove them - but I can load this project without issue otherwise, and Ionide recognizes all the package references and whatnot. Do you have a minimal reproduction that can be shared that doesn't involve removing all the file items?
> 
> I'm using the same .NET SDK and Ionide version on macOS.

Hi @cartermp,

thanks for picking up this. We have very mixed experience with Ionide. Some of us are working on Linux and some of us are on MacOS.  When we open our projects there is roughly a 10% chance that Ionide is loading properly.

Here are some screenshots:

These are different stages of non functioning Ionide:

![Screenshot 2022-02-03 at 16 56 52](https://user-images.githubusercontent.com/20079/152379463-690de897-a2ab-49d4-bb18-e0205cbe8f9f.png)


![Screenshot 2022-02-03 at 16 57 18](https://user-images.githubusercontent.com/20079/152379454-9121d746-d789-4d24-b259-dd9c7926e8ce.png)


Do you have a recommended way of debugging this further? Without a functional Ionide the development experience is not that great. We are also experimenting with Rider and VS 2022 preview to have a stable working environment for F#.

**Comment by cartermp** [comment]
@l1x let's separate out concerns here. Do you have a reproduction that leads to the notification you saw with that error message (assuming this involves opening a particular file...)? I can't find a way to reproduce that.

**Comment by l1x** [comment]
> @l1x let's separate out concerns here. Do you have a reproduction that leads to the notification you saw with that error message (assuming this involves opening a particular file...)? I can't find a way to reproduce that.

Ok.

```
Invalid attempt to spread non-iterable instance. In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
Cannot read property 'subscriptions' of undefined
```

I get this whenever I open my project.

Project build:

```
> dotnet build -nologo
  Determining projects to restore...
  All projects are up-to-date for restore.
  depoxy -> /Users/l1x/code/datadeft/depoxy/backend/depoxy/bin/Debug/net6.0/depoxy.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:06.62
```

Executable runs:

```
> dotnet bin/Debug/net6.0/depoxy.dll
Starting up...
Updating cache...
```

depoxy.fsproj

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <RootNamespace>DataDeft.Depoxy</RootNamespace>
    <OutPutType>Exe</OutPutType>
    <TargetFramework>net6.0</TargetFramework>
    <LangVersion>latest</LangVersion>
	  <ImplicitUsings>enable</ImplicitUsings>
    <AWSProjectType>Lambda</AWSProjectType>
    <PublishTrimmed>true</PublishTrimmed>
	  <PublishReadyToRun>true</PublishReadyToRun>
    <AssemblyName>depoxy</AssemblyName>
    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="List.fs" />
    <Compile Include="Seq.fs" />
    <Compile Include="Option.fs" />
    <Compile Include="Result.fs" />
    <!-- config -->
    <Compile Include="Config.fs" />
    <!-- lib -->
    <Compile Include="Jwt.fs" />
    <Compile Include="Hashing.fs" />
    <Compile Include="Utils.fs" />
    <Compile Include="Athena.fs" />
    <Compile Include="Glue.fs" />
    <Compile Include="Ses.fs" />
    <Compile Include="S3.fs" />
    <Compile Include="ExternalId.fs" />
    <!-- init -->
    <Compile Include="Init.fs" />
    <!-- api:common -->
    <Compile Include="ApiCommon.fs" />
    <!-- api:specific -->
    <Compile Include="UserApi.fs" />
    <Compile Include="JobApi.fs" />
    <Compile Include="TableApi.fs" />
    <Compile Include="AggregateApi.fs" />
    <!-- handler -->
    <Compile Include="Handler.fs" />
    <!-- main -->
    <Compile Include="Main.fs" />

    <None Include="config\**\*.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    </None>

  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Amazon.Lambda.AspNetCoreServer" Version="7.0.1" />
    <PackageReference Include="Amazon.Lambda.Core" Version="2.1.0" />
    <PackageReference Include="Amazon.Lambda.Serialization.Json" Version="2.0.0" />
    <PackageReference Include="AWSSDK.Glue" Version="3.7.13" />
    <PackageReference Include="AWSSDK.SimpleEmailV2" Version="3.7.1.46" />
    <PackageReference Include="AWSSDK.Athena" Version="3.7.1" />
    <PackageReference Include="AWSSDK.S3" Version="3.7.5" />
    <PackageReference Include="FSharp.Data" Version="4.2.6" />
    <PackageReference Include="Falco" Version="3.1.10" />
    <PackageReference Include="FSharp.SystemTextJson" Version="0.17.4" />
    <PackageReference Include="LazyCache" Version="2.4.0" />
    <PackageReference Include="SimpleBase" Version="3.1.0" />
    <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" Version="6.15.1" />
    <PackageReference Include="System.Data.HashFunction.SpookyHash" Version="2.0.0" />
  </ItemGroup>

  <ItemGroup>
      <FrameworkReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>

</Project>
```

depoxy.sln

```
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio Version 16
VisualStudioVersion = 16.0.30114.105
MinimumVisualStudioVersion = 10.0.40219.1
Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "depoxy", "depoxy.fsproj", "{5FDAEC48-D5A9-494B-9BB2-CABF722F25F2}"
EndProject
Global
        GlobalSection(SolutionConfigurationPlatforms) = preSolution
                Debug|Any CPU = Debug|Any CPU
                Release|Any CPU = Release|Any CPU
        EndGlobalSection
        GlobalSection(SolutionProperties) = preSolution
                HideSolutionNode = FALSE
        EndGlobalSection
        GlobalSection(ProjectConfigurationPlatforms) = postSolution
                {5FDAEC48-D5A9-494B-9BB2-CABF722F25F2}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
                {5FDAEC48-D5A9-494B-9BB2-CABF722F25F2}.Debug|Any CPU.Build.0 = Debug|Any CPU
                {5FDAEC48-D5A9-494B-9BB2-CABF722F25F2}.Release|Any CPU.ActiveCfg = Release|Any CPU
                {5FDAEC48-D5A9-494B-9BB2-CABF722F25F2}.Release|Any CPU.Build.0 = Release|Any CPU
        EndGlobalSection
EndGlobal
```

dotnet info:

```
> dotnet  --info
.NET SDK (reflecting any global.json):
 Version:   6.0.101
 Commit:    ef49f6213a

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  11.6
 OS Platform: Darwin
 RID:         osx.11.0-arm64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.101/

Host (useful for support):
  Version: 6.0.1
  Commit:  3a25a7f1cc

.NET SDKs installed:
  6.0.100 [/usr/local/share/dotnet/sdk]
  6.0.101 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

What else can I give you or what else should I do to have debug messages where this setup produces that error?

**Comment by cartermp** [comment]
Three things would be great:

1. Specific steps that lead to this ("open project" can mean a lot of things)
2. A minimally reproducible project that also shows this, if possible. I don't have access to your codebase, so I can't reproduce this to see what's going on.
3. Output of getting logs `F#: Get Ionide logs` from command palette

**Comment by larjo** [comment]
I got the same error just by starting vscode from WSL2 and running the command "FSI: Start".
![image](https://user-images.githubusercontent.com/1218329/152492884-f266ca52-0b29-4c05-9411-5b45e75a5f38.png)

The error went away when I downgraded the Ionide extension. And it still worked when I reinstalled the latest version (5.10.2).

So you could try to reinstall Ionide. (And make sure to restart vscode).

**Comment by l1x** [comment]
> I got the same error just by starting vscode from WSL2 and running the command "FSI: Start". ![image](https://user-images.githubusercontent.com/1218329/152492884-f266ca52-0b29-4c05-9411-5b45e75a5f38.png)
> 
> The error went away when I downgraded the Ionide extension. And it still worked when I reinstalled the latest version (5.10.2).
> 
> So you could try to reinstall Ionide. (And make sure to restart vscode).
> 
> I am running: WSL2 (Ubuntu 21.10) .NET SDK 6.0.101

Thanks @larjo. It is our daily routine to uninstall and reinstall Ionide.

**Comment by Krzysztof-Cieslak** [comment]
Can we stay on topic without passive-aggressive statements like "It is our daily routine to uninstall and reinstall Ionide.". Thanks.

**Comment by Wallace-Kelly** [comment]
For what it is worth, about half of my team is getting this error. The F# icon does not appear when we open the folder with our project. We will continue to try to figure it out, but welcome suggestions.
![image](https://user-images.githubusercontent.com/52675700/154096141-04a8f805-f449-4329-b80e-a7acb016adad.png)

**Comment by cartermp** [comment]
Thanks @Wallace-Kelly - unfortunately I can't reproduce this one as per your steps. My immediate thinking is that there's some oddity involving another extension that I don't have installed, which could explain why I don't see this issue. Could you list the extensions you have installed, if possible?

**Comment by Wallace-Kelly** [comment]
Thank you, @cartermp for trying. When I deleted my `~\.vscode` folder, it effectively removed all my extensions. At the moment I only have these installed:

* Ionide-fsharp, Version: 5.10.2
* ms-dotnettools.csharp, Version: 1.24.0

Here is some version information

```
> dotnet --version
6.0.101

> dotnet fsi --version
Microsoft (R) F# Interactive version 12.0.0.0 for F# 6.0

> dotnet --list-sdks
2.1.701 [C:\Program Files\dotnet\sdk]
2.2.401 [C:\Program Files\dotnet\sdk]
3.1.120 [C:\Program Files\dotnet\sdk]
3.1.416 [C:\Program Files\dotnet\sdk]
5.0.104 [C:\Program Files\dotnet\sdk]
5.0.211 [C:\Program Files\dotnet\sdk]
6.0.101 [C:\Program Files\dotnet\sdk]
```

Even after a machine reboot, I'm not seeing the F# icon in the Activity Bar.

![image](https://user-images.githubusercontent.com/52675700/154140589-55ef8510-4693-4ec8-80b7-290aae8bee02.png)

I tried running the "F#: Get info for diagnostics" command. It resulted in an error box:

![image](https://user-images.githubusercontent.com/52675700/154141310-9cad249e-8f39-4d8c-a8d1-689f6e220ce4.png)

```
[2022-02-15 15:11:41.033] [exthost] [error] {"type":"Buffer","data":[67,111,117,108,100,32,110,111,116,32,101,120,101,99,117,116,101,32,98,101,99,97,117,115,101,32,116,104,101,32,115,112,101,99,105,102,105,101,100,32,99,111,109,109,97,110,100,32,111,114,32,102,105,108,101,32,119,97,115,32,110,111,116,32,102,111,117,110,100,46,13,10,80,111,115,115,105,98,108,101,32,114,101,97,115,111,110,115,32,102,111,114,32,116,104,105,115,32,105,110,99,108,117,100,101,58,13,10,32,32,42,32,89,111,117,32,109,105,115,115,112,101,108,108,101,100,32,97,32,98,117,105,108,116,45,105,110,32,100,111,116,110,101,116,32,99,111,109,109,97,110,100,46,13,10,32,32,42,32,89,111,117,32,105,110,116,101,110,100,101,100,32,116,111,32,101,120,101,99,117,116,101,32,97,32,46,78,69,84,32,112,114,111,103,114,97,109,44,32,98,117,116,32,100,111,116,110,101,116,45,109,115,98,117,105,108,100,32,47,118,101,114,115,105,111,110,32,100,111,101,115,32,110,111,116,32,101,120,105,115,116,46,13,10,32,32,42,32,89,111,117,32,105,110,116,101,110,100,101,100,32,116,111,32,114,117,110,32,97,32,103,108,111,98,97,108,32,116,111,111,108,44,32,98,117,116,32,97,32,100,111,116,110,101,116,45,112,114,101,102,105,120,101,100,32,101,120,101,99,117,116,97,98,108,101,32,119,105,116,104,32,116,104,105,115,32,110,97,109,101,32,99,111,117,108,100,32,110,111,116,32,98,101,32,102,111,117,110,100,32,111,110,32,116,104,101,32,80,65,84,72,46,13,10]} fsharp.diagnostics.getInfos {"value":"Ionide.Ionide-fsharp","_lower":"ionide.ionide-fsharp"}
```

Let me know if I can help in any other way.

**Comment by Wallace-Kelly** [comment]
Doh! 🤦🏻‍♂️ I just figured out why the F# icon was not appearing in the left Activity Bar.

Visual Studio Code (version 1.64) introduced [a new feature called "Side Panel."](https://visualstudiomagazine.com/articles/2022/02/08/vs-code-1-64.aspx) Last week I had moved the F# Solution Explorer over to the right-side Side Panel to play with that feature. The problem is that Side Panel apparently does not re-open the next time I start Visual Studio Code. It did not open, I forgot about that new feature, and I could not find the F# Solution Explorer. Apparently, I'm hitting a bug/usability issue with this new Side Panel feature, not the Ionide extension.

How did I find this? I selected "F#: Focus on Solution Explorer View" and the right-side Side Panel appeared with the F# Solution Explorer in it.

![image](https://user-images.githubusercontent.com/52675700/154144116-bc4b565e-e8ad-4eaa-9f22-781d6becf1ed.png)

All these other error messages I reported are likely unrelated. 

My apologies!
